### DISCUSSION TITLE
how to foramtter display path?

### DISCUSSION THREAD

**Discussion by mycf** [ibhagwan/fzf-lua#1227] [Q&A] [answered]
How can I do it in `fzflua`? like I`ve custom implemented it in Telescope The following
this is a raw path
```
jdt:/contents/mybatis-3.5.14.jar/org.apache.ibatis.plugin/Invocation.class?=spring-pz-consumer/%5C/Users%5C/ycf%5C/.m2%5C/repository%5C/org%5C/mybatis%5C/mybatis%5C/3.5.14%5C/mybatis-3.5.14.jar=/maven.pomderived=/true=/=/maven.groupId=/org.mybatis=/=/maven.artifactId=/mybatis=/=/maven.version=/3.5.14=/=/maven.scope=/compile=/=/maven.pomderived=/true=/%3Corg.apache.ibatis.plugin(Invocation.class"
```
lsp_work_symbol
<img width="528" alt="image" src="https://github.com/ibhagwan/fzf-lua/assets/36999825/4f0e0935-2fc1-4db6-b1a3-d5b2abda7a76">
lsp_reference
<img width="1111" alt="image" src="https://github.com/ibhagwan/fzf-lua/assets/36999825/e9c6649e-9543-4102-85a5-6aedcc94061b">
lsp_implement
<img width="537" alt="image" src="https://github.com/ibhagwan/fzf-lua/assets/36999825/12bb8507-956b-46c1-bdb0-8270620cc6aa">

**Comment by ibhagwan** [comment, marked as answer]
Fzf-lua uses an external process, therefore all the data has to be contained within the line, for what you’re after you’ll  need a custom formatter that duplicates the full path/symbol data and then hide the metadata with `--with-nth`.

You might be able to hack your way with `_fmt.to` and `_fmt.from`:
https://github.com/ibhagwan/fzf-lua/blob/b92220ec838c195eb1c711daa69c905b1d7b8d8c/lua/fzf-lua/defaults.lua#L752-L758

https://github.com/ibhagwan/fzf-lua/blob/b92220ec838c195eb1c711daa69c905b1d7b8d8c/lua/fzf-lua/defaults.lua#L744-L750

**Reply by mycf** [reply]
hi, I've tried the `to` function, why only get part of string
```lua
          to = function(s, _)
            print(s)
            return "web"
          end,
```
<img width="1132" alt="image" src="https://github.com/ibhagwan/fzf-lua/assets/36999825/73b32966-b84a-46d3-bd43-17a34d1bb20f">
<img width="414" alt="image" src="https://github.com/ibhagwan/fzf-lua/assets/36999825/b2b298be-c56b-436d-97e4-6a70390e823b">

**Reply by ibhagwan** [reply]
I’m not sure, you’d have to dig into the code for that, it’s possible that for symbols the `to` is already “baked in” for performance (as I didn’t want to transform twice), as mentioned here:
https://github.com/ibhagwan/fzf-lua/blob/b92220ec838c195eb1c711daa69c905b1d7b8d8c/lua/fzf-lua/defaults.lua#L744-L750

Formatting and customization is endless and thus is always a slow improvement process of interfaces/API but it’s not urgent as most users just use the provided defaults and builtin customizations.
